gwcs/wcs.py

```diff
@@ -634,3 +641,269 @@ def fix_inputs(self, fixed):
         new_pipeline.append((step0[0], new_transform))
         new_pipeline.extend(self.pipeline[1:])
         return self.__class__(new_pipeline)
+
+    def to_fits_sip(self, bounding_box, max_pix_error=0.25, degree=None,
+                    max_inv_pix_error=0.25, inv_degree=None, 
+                    npoints=32, verbose=False):
+        """
+        Construct a SIP-based approximation to the WCS in the form of a FITS header
+
+        This assumes a tangent projection.
+
+        The default mode in using this attempts to achieve roughly 0.25 pixel
+        accuracy over the whole image.
+
+        Parameters
+        ----------
+        bounding_box : a pair of tuples, each consisting of two integers
+            Represents the range of pixel values in both dimensions
+            ((xmin, xmax), (ymin, ymax))
+        max_pix_error : float, optional
+            Maximum allowed error over the domain of the pixel array. This
+            error is the equivalent pixel error that corresponds to the maximum
+            error in the output coordinate resulting from the fit based on
+            a nominal plate scale.
+        degree : int, optional
+            Degree of the SIP polynomial. If supplied, max_pixel_error is ignored.
+        max_inv_error : float, optional
+            Maximum allowed inverse error over the domain of the pixel array
+            in pixel units. If None, no inverse is generated.
+        inv_degree : int, optional
+            Degree of the inverse SIP polynomial. If supplied max_inv_pixel_error
+            is ignored.
+        npoints : int, optional
+            The number of points in each dimension to sample the bounding box
+            for use in the SIP fit.
+        verbose : bool, optional
+            print progress of fits
+
+        Returns
+        -------
+        FITS header with all SIP WCS keywords
+
+        Raises
+        ------
+        ValueError
+            If the WCS is not 2D, an exception will be raised. If the specified accuracy
+            (both forward and inverse, both rms and maximum) is not achieved an exception
+            will be raised.
+
+        Notes
+        -----
+
+        Use of this requires a judicious choice of required accuracies. Attempts to use
+        higher degrees (~7 or higher) will typically fail due floating point problems
+        that arise with high powers.
+
+
+        """
+        if (self.forward_transform.n_inputs != 2
+                or self.forward_transform.n_outputs != 2):
```

Comment:
I wonder about this condition.
- Perhaps it's good to relax the check for the number of inputs, as >= 2, to allow for choosing a subsection of an image parametrized by an additional quantity. An example is the NIRISS AMI mode where different parts of the image will have different distortions. (Hm, I wonder how they are going to represent that with FITS).
Or perhaps not check for number of inputs.
- The number of outputs should certainly be 2 but this doesn't mean it's an image. There are a couple of ways to do this:
`wcsobj.world_axis_physical_types` should be a tuple of length 2. My understanding is that in most cases each item should start with `pos` as in `('pos.eq.ra', 'pos.eq.dec')`. In some cases it can start with `custom`.
Perhaps the most robust way is to check that the output frame is a Celestial frame:
```
from gwcs import coordinate_frames as cf 
isinstance(wcsobject.output_frame, cf.CelestialFrame)
```